src/main/scala/org/graphframes/GraphFrame.scala

```diff
@@ -475,13 +474,17 @@ class GraphFrame private(
       indexedVertices.select(
         col(ATTR + "." + ID).cast("long").as(LONG_ID), col(ATTR + "." + ID).as(ID), col(ATTR))
     } else {
-      val withLongIds = vertices.select(ID)
-        .repartition(col(ID))
-        .distinct()
-        .sortWithinPartitions(ID)
-        .withColumn(LONG_ID, monotonically_increasing_id())
-        .persist(StorageLevel.MEMORY_AND_DISK)
-      vertices.select(col(ID), nestAsCol(vertices, ATTR))
+      val repartitionedVertices = vertices.repartition(vertices.rdd.getNumPartitions, col(ID))
+      val idDF = repartitionedVertices.select(ID)
+      val idType = idDF.schema(0)
+      // DF monotonically_increasing_id() method assign id that are greater them max int,
```

Comment:
@artem-aliev 
zipWithIndex generates a Long index  isn't it?
For large graphs (>Max.INT vertices) you want to have unique ids. 
Can you elaborate?

Reply:
Right, this limitation is still exists. It is GraphX one. But I do not expect any of graph algorithms will finish in reasonable time with >Max.INT vertices.

Reply:
Well - disagree, we have regular graphs with 10s of Billions of vertices.
Adding this limitation of 2,147,483,647 vertices will just limit the graphframes usage